### PR TITLE
build with gha

### PR DESCRIPTION
**Pull request by georgeblahblah**
## What does this change?

- Builds the project with GitHub Actions
- Uploads to RiffRaff using `actions/riff-raff`
- ⚠️ I've taken this opportunity to move this project to standard RiffRaff naming, making the RiffRaff project now called `membership-about-us::about-us`

## Deployment checklist - before merge (ordered)

- [x] Pause the [`About us` project in TeamCity](https://teamcity.gutools.co.uk/project/memsub_AboutUs?mode=builds#all-projects)
- [x] Remove the (now unused) `artifact.json` file
- [x] Remove the `ci.sh` file, originally used by TeamCity
- [x] Add a deployment restriction to the [old About Us project in RiffRaff](https://riffraff.gutools.co.uk/deployment/history?projectName=About%20us&page=1)
- [x] Disable the [Continuous Deployment configuration ](https://riffraff.gutools.co.uk/deployment/continuous/0b7854a7-a064-4fc5-b517-bd0a1e833293/edit) for the old project in RiffRaff
- [x] Add a [Continuous Deployment configuration](https://riffraff.gutools.co.uk/deployment/continuous) for the new Riff Raff project, so any builds on the `main` branch are automatically deployed to PROD

## How to test

- A sample deployment to CODE was successful: https://gu-about-us.s3-eu-west-1.amazonaws.com/CODE/gu-about-us-upload/about/index.html
  - [RiffRaff deploy](https://riffraff.gutools.co.uk/deployment/view/f8be3890-53ef-48f8-aa07-5220db9341c2)

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          aws-region: eu-west-1
+          role-to-assume: ${{ secrets.GU_RIFF_RAFF_ROLE_ARN }}
+
+      - run: npm ci
+      - run: npm run build
+      - run: |
+          mkdir about
+          mv out/* about
+          mv about out
```
Comment (AshCorr) [inline .github/workflows/deploy.yml:33]: Am I understanding this correctly that we

1. Create an about folder
2. Move the contents of `out/` to `about/`
3. Move about into `out/`
4. The final folder structure is `out/about/out`

**Review by AshCorr — APPROVED** [review]
👏 👏 👏

**Review by ioannakok — APPROVED** [review]
Nice one 👏

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (georgeblahblah) [reply .github/workflows/deploy.yml:33]: I think it's doing 3 steps:

1:
```
out/
  index.html
```

2:
```
out/
  #nothing in here now
about/
  index.html
```

3:
```
out/
  about/
    index.html
```

The important thing is that we need to upload a directory called "about" to S3, which includes the built application. May be a simpler way to do this, I'll play around!

Comment (georgeblahblah) [reply .github/workflows/deploy.yml]: I've updated the next config to output directly to `out/about` so we don't have to move files around

**Pull request merged**